Sources/PIALibrary/Common/Data/Networking/NetworkRequestClient.swift

```diff
@@ -12,31 +12,15 @@ protocol NetworkRequestClientType {
 class NetworkRequestClient: NetworkRequestClientType {
     private let networkConnectionRequestProvider: NetworkConnectionRequestProviderType
     private let endpointManager: EndpointManagerType
-    private let refreshAuthTokensChecker: RefreshAuthTokensCheckerType
     
-    init(networkConnectionRequestProvider: NetworkConnectionRequestProviderType, endpointManager: EndpointManagerType, refreshAuthTokensChecker: RefreshAuthTokensCheckerType) {
+    init(networkConnectionRequestProvider: NetworkConnectionRequestProviderType, endpointManager: EndpointManagerType) {
         self.networkConnectionRequestProvider = networkConnectionRequestProvider
         self.endpointManager = endpointManager
-        self.refreshAuthTokensChecker = refreshAuthTokensChecker
     }
     
     func executeRequest(with configuration: NetworkRequestConfigurationType, completion: @escaping Completion) {
 
-        if configuration.refreshAuthTokensIfNeeded {
-            // 1. Refresh the auth tokens before executing the request
-            refreshAuthTokensChecker.refreshIfNeeded { error in
-                if let error {
-                    // The request fails if refreshing the tokens have failed
-                    completion(error, nil)
-                } else {
-                    // 2. Execute the request
-                    self.startRequest(with: configuration, completion: completion)
-                }
-            }
-        } else {
-            startRequest(with: configuration, completion: completion)
-        }
-        
+        startRequest(with: configuration, completion: completion)
```

Comment:
This is the root cause  of the circular dependency: The `RefreshAuthTokensCheckerType` was having a dependency on `RefreshApiTokenUseCase` and `RefreshVpnTokenUseCase` that at the same time, were dependent on `NetworkRequestClient`